sdk/tests/tutorials/ibor/test_reference_portfolios.py

```diff
@@ -33,16 +34,26 @@ def setUpClass(cls):
         cls.instrument_loader = InstrumentLoader(cls.instruments_api)
         cls.instrument_ids = cls.instrument_loader.load_instruments()
 
+        cls.id_generator = IdGenerator(default_scope=TestDataUtilities.tutorials_scope)
+
+    @classmethod
+    def tearDownClass(cls):
+        for _, scope, code in cls.id_generator.pop_scope_and_codes():
+            try:
+                cls.portfolios_api.delete_portfolio(scope, code)
```

Comment:
Just for awareness, proliferation of data could happen here with error response codes,

Reply:
Its a good point, the idea is that the `IdGenerator` just generates ids, its the responsibly of the test to clean up. If there is an error during that (which _shouldn't_ happen but in reality could) it would be up to the test to tidy up. In reality I think if say the delete api call failed, it would most likely need a manual intervention.

Thoughts?

Reply:
Agreed. As long as we have data in an identifiable and separate scope, it should be relatively simple to manually delete data.